cloudsql/postgres/database-sql/run_tests

```diff
@@ -0,0 +1,9 @@
+#! /bin/sh
+
+# Run the tests with default values for all
+# environment variable
+go test -timeout $TIMEOUT -v "${1:-./...}" | tee sponge_log.log
+
+# Run the tests again with custom environment
+# variables that force a Unix connection
+DB_HOST=; go test -timeout $TIMEOUT -v "${1:-./...}" | tee sponge_log.log
```

Comment:
Go tests can set environment variables before calling whatever method is under test. In particular, you can unset `DB_HOST` and run the test, directly from the Go code.

In general, I'm a fan of as little Bash as possible. I'd also like to keep all of the `go test` calls in the same `system_test.sh` script so it's easier to keep them all consistent.

Reply:
I share your preference for minimizing bash and taking care of everything in Go. My initial thinking here was to solve two problems with the introduction of this script:

1) Unsetting the environment variable, which can move to the testing layer, but also
2) Run these tests twice, once with the variable set and once without.

Do you have any recommendations on how to preserve that second goal?

Reply:
That's usually done with [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests) in Go. Something like this:

```
tests := []struct{
  dbHost string
}{
  {dbHost: "8.8.8.8"},
  {},
}

for _, test := range tests {
  oldDBHost := os.Getenv("DB_HOST")
  os.Setenv(test.dbHost)
  // ...
  os.Setenv(oldDBHost)
}
```